app/design/frontend/Diana/default-theme/web/js/render-stores.js

```diff
@@ -0,0 +1,123 @@
+            self.renderPaginatedElements(self.availableStores());
+            self.availableStores.subscribe(function(value) {
+                if (!_.isUndefined(value)) {
+                    self.computeQuantityOfPages(self.availableStores());
+                    self.renderPaginatedElements(self.availableStores());
+                }
+            });
+        },
+        setInitialCurrentPageFromURLParameters: function () {
+            var self = this;
+            var url = new URL(window.location.href);
+            if (!_.isNull(url.href.match(/page=\d/))) {
+                self.paramPage(url.href.match(/page=\d/)[0].match(/\d/)[0]);
+                self.currentPageNumber(self.paramPage() - 1);
+            } else {
+                self.paramPage(null);
+                self.currentPageNumber(0);
+            }
+        },
+        computeQuantityOfPages: function (data) {
+            var self = this;
+            self.quantityOfPagesRenderer([]);
+            self.quantityOfPages = ko.computed(function() {
+                var quantityOfPages = Math.ceil(data.length / self.elementsPerPage);
+                return quantityOfPages;
+            });
+            for (var i = 1; i <= self.quantityOfPages(); i++) {
+                var isCurrent = (+self.currentPageNumber() + 1 == i);
+                self.quantityOfPagesRenderer.push({
+                    numberOfPage: i,
+                    isCurrent: ko.observable(isCurrent)
+                });
+            }
+        },
+        renderPaginatedElements: function(data) {
+            var self = this;
+            var firstPaginatedElement = self.currentPageNumber() * self.elementsPerPage;
+            self.paginatedElements([]);
+            if (self.quantityOfPages() === 1) {
+                for (var item of self.availableStores()) {
+                    self.paginatedElements.push(item)
+                }
+            } else {
+                var paginatedElements = data.slice(firstPaginatedElement, firstPaginatedElement + self.elementsPerPage);
+                for (var item of paginatedElements) {
+                    self.paginatedElements.push(item);
+                }
+            }
+        },
+        nextPage: function (element, event) {
+            var self = this;
+            var currentPageNumber = self.currentPageNumber();
+            if (currentPageNumber < self.quantityOfPages() - 1) {
+                self.currentPageNumber(currentPageNumber + 1);
+                self.renderPaginatedElements(self.availableStores());
+                self.quantityOfPagesRenderer()[currentPageNumber].isCurrent(false);
+                self.quantityOfPagesRenderer()[self.currentPageNumber()].isCurrent(true);
+                var url = new URL(window.location.href);
+                url.searchParams.set('page', `${currentPageNumber + 2}`);
+                window.history.replaceState(null, null, url);
+            }
+        },
+        previousPage: function() {
```

Comment:
@Dianaeva It's wiser to move functionality for `nextPage`, `previousPage` and `linkToPageNumber` to the separate method and call it with different arguments to avoid duplicate code